### ISSUE TITLE
SD_MMC: Incorrect pin settings for ESP32

### ISSUE THREAD

**Issue by broddo** [espressif/arduino-esp32#10709]
### Board

ESP32-DEVKITC-32E

### Device Description

SDCARD breakout board wired via breadboard

### Hardware Configuration

N/A

### Version
v3.1.0-RC2 804ee127
IDF v5.3.1

### IDE Name

VSCode

### Operating System

macOS 14.5

### Flash frequency

40 Mhz

### PSRAM enabled

yes

### Upload speed

460800

### Description

I recently updated one of my old projects that was using IDF v4 to IDF v5 and upgrade to the latest version of the Arduino library too. The hardware for this project includes an eMMC card that was functioning fine on the old libraries. 

The eMMC card no longer works when using the latest libraries. The reason is due to several calls to the `perimanSetPinBus` which result in this error: 

`[   177][E][esp32-hal-periman.c:122] perimanSetPinBus(): Invalid pin: 255` 

These calls fail because _pin_clk, _pin_cmd are indeed set to -1 if you are configuring for an ESP32. The solution is was to just add a `#ifndef CONFIG_IDF_TARGET_ESP32` to exclude these calls on ESP32 hardware. 

### Sketch

```cpp
#include <Arduino.h>
#include <SD_MMC.h>


void setup() {
  Serial.begin(115200);
  if (!SD_MMC.begin("/sdmmc")) {
    Serial.println("Failed to mount SDMMC module");
  }
}

void loop() { }
```

### Debug Message

```plain
[   177][E][esp32-hal-periman.c:122] perimanSetPinBus(): Invalid pin: 255
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by lbernstone** [comment]
In the constructor, [_pin_clk = SDMMC_SLOT1_IOMUX_PIN_NUM_CLK](https://github.com/espressif/arduino-esp32/blob/master/libraries/SD_MMC/src/SD_MMC.cpp#L49), so it should use the pins as defined in [soc/sdmmc_pins.h](https://github.com/espressif/esp-idf/blob/v5.3/components/soc/esp32/include/soc/sdmmc_pins.h).  What do you get if you print out SDMMC_SLOT1_IOMUX_PIN_NUM_CLK?

**Comment by broddo** [comment]
D'oh - I missed the define `BOARD_HAS_SDMMC`. Sorry - I can't believe I missed that. It wasn't in my project originally as I using an older version of this library that didn't require it and I'm running custom hardware. I see that it is defined in some board variants but can you suggest a good place for me to define this? 

(I'm not using platformio - I'm using IDF v5.3.1 with Arduino ESP32 (v 3.1.0-RC2 804ee127) added as a component.)

**Comment by me-no-dev** [comment]
It should not require `BOARD_HAS_SDMMC` to work. @P-R-O-C-H-Y PTAL

**Comment by P-R-O-C-H-Y** [comment]
Closing as solved by PR #10713
